data-prepper-plugins/opensearch/src/main/resources/index-template/metrics-otel-index-template.json

```diff
@@ -0,0 +1,243 @@
+{
+  "version": 1,
+  "template": {
+    "mappings": {
+      "dynamic_templates": [
+        {
+          "resources_map": {
+            "mapping": {
+              "type": "keyword"
+            },
+            "path_match": "resource.*"
+          }
+        }
+        ],
+      "_source": {
+        "enabled": true
+      },
+      "properties": {
+        "name": {
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword",
+              "ignore_above": 256
+            }
```

Comment:
This is aligned with [Simple Schema for Observability](https://github.com/opensearch-project/opensearch-catalog/tree/main/docs/schema/observability), see https://github.com/opensearch-project/opensearch-catalog/blob/0aee4fbc2df50557dd00b7f91a71099d47c56e3d/schema/observability/metrics/metrics-1.0.0.mapping#L55-L63

@dlvenable I consider this to be a feedback/review of the simple schema

Reply:
This is an interesting point. I see that CloudWatch has a 255 character limit. Perhaps when combined with the namespace to make a single name it would exceed that value.

https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html